Earthfile

```diff
@@ -467,23 +467,39 @@ base-image:
 
     RUN rm -rf /tmp/*
 
-    DO +CONTAINER_IMAGE_VERSION -VERSION=${OS_VERSION}
-    ARG _CIMG=$(cat IMAGE)
+    COPY ./naming.sh .
+    RUN ./naming.sh container_artifact_name
+    RUN ./naming.sh container_artifact_name > ./IMAGE
 
     # luet cleanup
     RUN luet cleanup
     RUN rm -rf /var/luet
 
+    ARG _CIMG=$(cat ./IMAGE)
     SAVE IMAGE $_CIMG
     SAVE ARTIFACT IMAGE AS LOCAL build/IMAGE
     SAVE ARTIFACT VERSION AS LOCAL build/VERSION
     SAVE ARTIFACT /etc/kairos/versions.yaml versions.yaml AS LOCAL build/versions.yaml
 
 image-rootfs:
+    ARG --required FAMILY
+    ARG --required FLAVOR
+    ARG --required BASE_IMAGE
+    ARG --required MODEL
+    ARG --required VARIANT
+
     FROM +base-image
+
     SAVE ARTIFACT --keep-own /. rootfs
 
 uki-artifacts:
+    ARG --required FAMILY # The dockerfile to use
```

Comment:
damn, cant we make this a global thing instead of having them in each target?

Reply:
I'd rather not. Anything we make global becomes accessible to every target, even those that don't need it. Then you can't make it `--required` either (since it's not required by all targets, right?). It looks ugly right now but I'm hoping on the next iteration, the only thing most targets will need is the a "base image" arg which will be a pre-built image. That one will be build directly with docker to ensure that we can always build with just `docker`, no earthly required.